azurerm/internal/services/machinelearning/machine_learning_compute_cluster_resource.go

```diff
@@ -0,0 +1,438 @@
+		// Description - The description of the Machine Learning compute.
+		Description: &description,
+		// CreatedOn - READ-ONLY; The date and time when the compute was created.
+		// ModifiedOn - READ-ONLY; The date and time when the compute was last modified.
+		// ResourceID - ARM resource id of the underlying compute
+		// ProvisioningErrors - READ-ONLY; Errors during provisioning
+		// IsAttachedCompute - READ-ONLY; Indicating whether the compute was provisioned by user and brought from outside if true, or machine learning service provisioned it if false.
+		// ComputeType - Possible values include: 'ComputeTypeCompute', 'ComputeTypeAKS1', 'ComputeTypeAmlCompute1', 'ComputeTypeVirtualMachine1', 'ComputeTypeHDInsight1', 'ComputeTypeDataFactory1', 'ComputeTypeDatabricks1', 'ComputeTypeDataLakeAnalytics1'
+		ComputeType: "ComputeTypeAmlCompute1",
+	}
+
+	amlComputeProperties, isAmlCompute := (machinelearningservices.BasicCompute).AsAmlCompute(computeClusterProperties)
+	if !isAmlCompute {
+		return fmt.Errorf("error: No Compute cluster")
+	}
+
+	computeClusterParameters := machinelearningservices.ComputeResource{
+		// Properties - Compute properties
+		Properties: amlComputeProperties,
+		// ID - READ-ONLY; Specifies the resource ID.
+		// Name - READ-ONLY; Specifies the name of the resource.
+		// Identity - The identity of the resource.
+		Identity: expandComputeClusterIdentity(identity),
+		// Location - Specifies the location of the resource.
+		Location: &location,
+		// Type - READ-ONLY; Specifies the type of the resource.
+		// Tags - Contains resource tags defined as key/value pairs.
+		Tags: tags.Expand(t),
+		// Sku - The sku of the workspace.
+		Sku: sku,
+	}
+
+	future, err := mlComputeClient.CreateOrUpdate(ctx, workspaceID.ResourceGroup, workspaceID.Name, name, computeClusterParameters)
+	if err != nil {
+		return fmt.Errorf("error creating Compute Cluster %q in workspace %q (Resource Group %q): %+v",
+			name, workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+	if err := future.WaitForCompletionRef(ctx, mlComputeClient.Client); err != nil {
+		return fmt.Errorf("error waiting for creation of Compute Cluster %q in workspace %q (Resource Group %q): %+v",
+			name, workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+	resp, err := mlComputeClient.Get(ctx, workspaceID.ResourceGroup, workspaceID.Name, name)
+	if err != nil {
+		return fmt.Errorf("error retrieving Compute Cluster Compute %q in workspace %q (Resource Group %q): %+v",
+			name, workspaceID.Name, workspaceID.ResourceGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("cannot read Compute Cluster ID %q in workspace %q (Resource Group %q) ID",
+			name, workspaceID.Name, workspaceID.ResourceGroup)
+	}
+
+	id, err := parse.ComputeClusterID(*resp.ID)
+	if err != nil {
+		return err
+	}
```

Comment:
As @tombuildsstuff suggested in the chat, since we have our parser and this is a new resource, we can build our own ID
```suggestion
	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
	id := parse.NewComputeClusterID(subscriptionId, workspaceID.ResourceGroup, workspaceID.Name, name)
```